Lib/test/test_builtin.py

```diff
@@ -832,6 +832,30 @@ class customdict(dict):  # this one should not do anything fancy
         self.assertRaisesRegex(NameError, "name 'superglobal' is not defined",
                                exec, code, {'__builtins__': customdict()})
 
+    def test_exec_builtins_mapping(self):
+        code = compile("superglobal", "test", "exec")
+        # works correctly
+        exec(code, {'__builtins__': types.MappingProxyType({'superglobal': 1})})
```

Comment:
As I wrote previously, I would prefer to check for the expected superglobal value here.

Reply:
Do you want to change also existing tests?